src/pinefarm/external/interface.py

```diff
@@ -33,12 +33,11 @@ def __init__(self, name, theory, pdf, timestamp=None):
         self.theory = theory
         self.pdf = pdf
         self.timestamp = timestamp
-
         if timestamp is None:
-            self.dest = tools.create_output_folder(self.name)
+            self.dest = tools.create_output_folder(self.name, self.theory)
```

Comment:
(it's the same complexity, here or there, but it is more isolated, since the function doesn't get the full theory and doesn't have to be aware that it has a key called `"ID"`)